### PR TITLE
drone: Add missing GOCACHE path for `make image-ci`

### LINKED ISSUE #7205: `docker:dind` floating ref upgrade broke Docker image builds
https://drone.gravitational.io/gravitational/teleport/13021/22/3

This appears to be something that works in `docker:20.10.6-dind` but not `docker:dind` (which uses 20.10.7)

Most likely an incorrect one-sided volume reference.

### PR DESCRIPTION
**Pull request by webvictim**
If the `GOCACHE` environment variable isn't set while running in Drone, the `-v` directive added to `DOCKERFLAGS` doesn't get properly formed. This was apparently OK in Docker 20.10.6, but results in container launch failures with Docker 20.10.7. This means that the Drone `build-docker-images` step fails: https://drone.gravitational.io/gravitational/teleport/13021/22/3

This also adds a sane default setting for `GOCACHE` if not set so we don't have to explicitly configure it inside every single Drone pipeline.

Fixes #7205

Backports required:
- `branch/v6`
- `branch/v6.2`

### CODE REVIEW

**Comment by webvictim** [comment]
Tag builds on `master` are still failing due to #7211. I'm going to fix that in a separate PR.

File: build.assets/Makefile [hunk 1]
```diff
@@ -50,6 +50,10 @@ ifeq ("$(DRONE)","true")
 UID := 1000
 GID := 1000
 NOROOT := -u 1000:1000
+# if running in CI and the GOCACHE environment variable is not set, set it to a sensible default
+ifeq ("$(GOCACHE)",)
+GOCACHE := /go/cache
```
Comment (awly) [inline build.assets/Makefile:55]: Let's use `go env GOCACHE` as the default

**Review by awly — APPROVED** [review]

File: .drone.yml [hunk 2]
```diff
@@ -3764,6 +3764,7 @@ steps:
     environment:
       UID: 1000
       GID: 1000
+      GOCACHE: /go/cache
```
Comment (wadells) [inline .drone.yml:3767]: Perhaps use this variable in the `mkdir -p` above as well?

**Review by quinqu — APPROVED** [review]

**Review by wadells — APPROVED** [review]

**Review by russjones — APPROVED** [review]
Bot.

**Comment by webvictim** [comment]
To follow up - using `go env GOCACHE` is actually undesirable, as many of the steps in `.drone.yml` that make use of `build.assets` targets use `docker` or `docker:git` base images which don't have `go` installed. Hardcoding the path is preferable.

**Pull request merged**